graylog-storage-opensearch2/src/main/java/org/graylog/storage/opensearch2/IndicesAdapterOS2.java

```diff
@@ -204,19 +212,32 @@ public Map<String, Object> getIndexMetaData(@Nonnull String index) {
     }
 
     @Override
-    public boolean ensureIndexTemplate(String templateName, Map<String, Object> template) {
-        final PutIndexTemplateRequest request = new PutIndexTemplateRequest(templateName)
-                .source(template);
+    public boolean ensureIndexTemplate(String templateName, Template template) {
+        var serializedMapping = serialize(template.mappings());
+        var settings = org.graylog.shaded.opensearch2.org.opensearch.common.settings.Settings.builder().loadFromMap(template.settings()).build();
+        var osTemplate = new org.graylog.shaded.opensearch2.org.opensearch.cluster.metadata.Template(settings, serializedMapping, null);
+        var indexTemplate = new ComposableIndexTemplate(template.indexPatterns(), osTemplate, null, template.order(), null, null);
+        var request = new PutComposableIndexTemplateRequest()
+                .name(templateName)
+                .indexTemplate(indexTemplate);
```

Comment:
Why don't you use `IndexTemplateAdapter` approach in OS codebase?

Reply:
I forgot :) Pushed that now.